pkg/kvstore/lock.go

```diff
@@ -38,22 +38,28 @@ type Lock struct {
 }
 
 var (
-	lockPathsMU lock.Mutex
+	lockPathsMU lock.RWMutex
 	lockPaths   = map[string]*lock.Mutex{}
 )
 
 // LockPath locks the specified path and returns the Lock
 func LockPath(path string) (*Lock, error) {
+	var (
+		localLock *lock.Mutex
+		ok        bool
+	)
```

Comment:
Not a big deal but why declare these here? you could `:=` below since we don't put things into the if line.

Reply:
Correct, thanks! Fixed